p2p/pb/proto_services.go

```diff
@@ -184,29 +183,30 @@ func DecodeQuaiResponse(respMsg *QuaiResponseMessage) (uint32, interface{}, erro
 // Converts a custom go type to a proto type and marhsals it into a protobuf message
 func ConvertAndMarshal(data interface{}) ([]byte, error) {
 	switch data := data.(type) {
-	case *types.WorkObject:
-		log.Global.Tracef("marshalling block: %+v", data)
-		protoBlock, err := data.ProtoEncode(types.BlockObject)
-		if err != nil {
-			return nil, err
+	case *types.WorkObject, *types.WorkObjectHeaderView, *types.WorkObjectBlockView:
+		var protoBlock *types.ProtoWorkObject
+		var err error
+		switch data := data.(type) {
+		case *types.WorkObject:
+			protoBlock, err = data.ProtoEncode(types.BlockObject)
```

Comment:
why would be encode it all as BlockObject? this should change based on the data right? wasnt that the whole point. I am referring to the case statements below

Reply:
This wasn't a decision introduced in this PR, in the previous code everything was being processed the same way, as a block.